File: Year 2/Semester 4/Programming and Projection Enironments/Week 6/ASaseaSaptamana/ASaseaSaptamana/domain/ShowDTO.cs
using ASaseaSaptamana.domain;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace ASaseaSaptamana.domain
{
    public class ShowDTO<ID> : Entity<ID>
    {
        private string title;
        private string date;
        private string place;
        private int available;
        private int unavailable;
        private int idArtist;

        public ShowDTO(ID id, string title, string date, string place, int available, int unavailable, int idArtist)
        :base(id)
        {
            this.title = title;
            this.date = date;
            this.place = place;
            this.available = available;
            this.unavailable = unavailable;
            this.idArtist = idArtist;
        }

        public string Title { get { return title; } }
        public string Date { get { return date; } }
        public string Place { get { return place; } }
        public int Available { get { return available;} }
        public int Unavailable { get {  return unavailable;} }
        public int IDArtist { get {  return idArtist; } }
    }
}

File: Year 2/Semester 4/Programming and Projection Environments/Week 5/JavaFifthWeek/src/main/java/com/example/temasaptamana5/domain/ShowDTO.java
package com.example.temasaptamana5.domain;

import java.time.LocalDate;

public class ShowDTO extends Entity<Integer>{
    private final String title;
    private final LocalDate date;
    private final String place;
    private final Integer available;
    private final Integer unavailable;
    private final Integer idArtist;

    public ShowDTO(Integer id, String title, LocalDate date, String place,
                   Integer available, Integer unavailable, Integer idArtist) {
        super(id);
        this.title = title;
        this.date = date;
        this.place = place;
        this.available = available;
        this.unavailable = unavailable;
        this.idArtist = idArtist;
    }

    public String getTitle() {
        return title;
    }

    public LocalDate getDate() {
        return date;
    }

    public String getPlace() {
        return place;
    }

    public Integer getAvailable() {
        return available;
    }

    public Integer getUnavailable() {
        return unavailable;
    }

    public Integer getIdArtist() {
        return idArtist;
    }
}
